doc/configuration.rst

```diff
@@ -428,40 +432,38 @@ You can create a custom sort key callable for the following configurations:
 * :ref:`subsection_order <sub_gallery_order>`
 * :ref:`minigallery_sort_order <minigallery_order>`
 
-When creating a custom sort key callable, you must ensure that the ``__repr__``
-is stable across runs. Sphinx determines if the build environment has
-changed, and thus if *all* documents should be rewritten, by examining the
-config values using ``md5(str(obj).encode()).hexdigest()`` in
-``sphinx/builders/html.py``. Default class instances in Python have their
-memory address in their ``__repr__`` which is why generally the ``__repr__``
-changes in each build.
+The best way to do this is to make use of a single instance of each class you want
+to use in some fully resolvable path, and then tell sphinx-gallery to resolve the
+callable from the fully qualified name. For example, you could write a function
```

Comment:
This does not seem to marry with the `CHANGES.rst` notes:

> 1. ``"mymod.utils.reset_argv"`` where you define a ``class _ResetArgv`` class in
   ``mymod.utils`` and instantiate it in your ``conf.py`` as
   ``reset_argv = _ResetArgv()``.

Reply:
Nice catch, fixed!